### DISCUSSION TITLE
CPU usage by thread

### DISCUSSION THREAD

**Discussion by GuillaumeEgret** [YueChen-C/py-ios-device#126] [Ideas]
Hello ! Is it currently possible to get the CPU usage per thread of an application ?

**Comment by YueChen-C** [comment]
Have you seen it in instrument or xcode monitoring?

**Reply by GuillaumeEgret** [reply]
Yes, with Xcode instruments with CPU Profiler or Time profiler you can have the details of the threads of a process with their CPU usage

**Reply by GuillaumeEgret** [reply]
Hello, quick update on this subject, here is a screenshot of Xcode Instruments with cpu usage per thread :
<img width="1440" alt="Screenshot 2023-10-30 at 18 17 16" src="https://github.com/YueChen-C/py-ios-device/assets/126566691/ca456a1e-923d-4e28-b044-6c7016c72266">

I would like to try to do this PR if it's not currently possible. Could you give me some pointers on how to do it ?

**Comment by YueChen-C** [comment]
This feature is not implemented and requires complex reverse engineering work

As far as I know, this module should use the `com.apple.instruments.server.services.coreprofilesessiontap` service.
But I did not do a deeper analysis of the relevant data flow

**Reply by GuillaumeEgret** [reply]
Thank you I'll try to look into it !
